### ISSUE TITLE
Add action for asynchronous tasks in browser?

### ISSUE THREAD

**Issue by awlayton** [johntitus/node-horseman#108]
For one of my project I ended up adding a function to my horseman that was similar to `.evaluate`, but which passed a node-style callback to the function (similar to `.do` but for functions run in the browser instead of in node). For example:
```javascript
horseman
    .evaluate2(function sumbitForm(done) {
        'use strict';

        try {
            var form = $('#rostform1').get(0);
            var method = (form.method || 'GET').toLowerCase();

            $[method](form.action, $(form).serialize(), function(data) {
                return done(null, data);
            });
        } catch (err) {
            return done(err);
        }
    })
    .then(function(data) { /* handle form response data */ })
    .catch(function(err) { /* something went wrong with submitting form */ });
```
I personally have found this action quite useful.

Should something like this be included in horseman? If so, it needs a better name than `.evaluate2`. Also, the implementation of it is a bit gross but it works.

**Comment by awlayton** [comment]
I would make a pull request, but I was not sure if it should be again master of v3.0.

**Comment by awlayton** [comment]
I supposed I could just do argument counting to detect when the function has a callback at the end and enhance evaluate to do my thing in that case, rather than making a new action.

Do you have any thoughts on this @johntitus?

**Comment by awlayton** [comment]
Also, this `.evaluate2` (or whatever it's called) rejects the Promise as expected if the function throws. This  makes things more like what is expected w.r.t #130.

**Comment by awlayton** [comment]
I am leaning toward's just having `.evaluate` do argument counting, rather than making a separate `.evaluate*` action that always passes a callback.

If anyone is opposed to that I would be happy to hear why.

**Comment by awlayton** [comment]
I have merged this into v3 as a part of addressing #77.
